tests/e2e_tests/cypress/plugins/index.js

```diff
@@ -5,37 +5,81 @@ const fs = require('fs');
+  let connectConfig = JSON.parse(connectConfigFile);
+  connectConfig.ServerURL = serverUrl;
+  if (token) {
+    connectConfig.ServerCertificate = '/certs/hosted.pem';
+    connectConfig.TenantToken = token;
+  }
+  fs.writeFileSync(`${projectRoot}/dockerClient/${fileName}-test.json`, JSON.stringify(connectConfig));
+};
+
 module.exports = (on, config) => {
   addMatchImageSnapshotPlugin(on, config);
   on('task', {
+    log(message) {
+      console.log(message);
+      return null;
+    },
     startClient({ token, count }) {
       let args = [
-        `-backend=${process.env.SERVER_URL}`,
         '-inventory="device_type:qemux86-64,client_version:mender-2.2.0,artifact_name:release-v1,kernel:test Linux version,mac_enp0:12.34"',
         '-invfreq=5',
         '-pollfreq=5',
-        `-count=${count}`
+        `-count=${count}`,
+        `-backend=${config.baseUrl.replace(/\/$/, '')}`
       ];
       if (token) {
-        console.log(`starting with: ${token}`);
         args.push(`-tenant=${token}`);
       }
+      console.log(`starting using: ./mender-stress-test-client ${args.join(' ')}`);
       let child = spawn('./mender-stress-test-client', args);
       child.on('error', err => console.error(`${err}`));
+      child.on('message', err => console.error(`${err}`));
+      child.on('spawn', err => console.error(`${err}`));
+      // child.stdout.on('data', data => {
+      //   console.log(`stdout mstc: ${data}`);
+      // });
+      child.stderr.on('data', data => {
+        console.error(`stderr mstc: ${data}`);
+      });
+      child.on('close', code => {
+        console.log(`child process exited with code ${code}`);
+      });
+
       return null;
     },
     startDockerClient({ token }) {
-      const connectConfigFile = fs.readFileSync('dockerClient/mender-connect.json');
-      let connectConfig = JSON.parse(connectConfigFile);
-      connectConfig.ServerURL = process.env.SERVER_URL || 'https://docker.mender.io';
-      fs.writeFileSync('dockerClient/mender-connect-test.json', JSON.stringify(connectConfig));
-      let args = ['run', '-v dockerClient/mender-connect-test.json:/etc/mender/mender-connect.conf', 'mendersoftware/mender-client-docker:connect'];
-      if (token) {
-        console.log(`starting with: ${token}`);
-        args.splice(1, 0, `-e TENANT_TOKEN=${token}`);
-      }
+      const baseUrl = config.baseUrl.replace(/\/$/, '');
+      updateConfigFileWithUrl('mender', baseUrl, token, config.projectRoot);
+      updateConfigFileWithUrl('mender-connect', baseUrl, token, config.projectRoot);
+      let args = [
+        'run',
+        '-v',
+        `${config.projectRoot}/hosted.pem:/certs/hosted.pem`,
+        '-v',
+        `${config.projectRoot}/dockerClient/mender-test.json:/etc/mender/mender.conf`,
+        '-v',
+        `${config.projectRoot}/dockerClient/mender-connect-test.json:/etc/mender/mender-connect.conf`,
+        'mendersoftware/mender-client-docker:connect'
```

Comment:
so, `:connect` will always exists, right? maybe lets make some variable of it if not a parameter.

Reply:
it's built in an earlier job for that specific purpose (running it in the e2e tests to test the OS setup + in these tests against staging)